paasta_tools/mesos_maintenance.py

```diff
@@ -331,7 +396,110 @@ def load_credentials(mesos_secrets='/nail/etc/mesos-slave-secret'):
         log.error("%s does not contain Mesos slave credentials in the expected format. "
                   "See http://mesos.apache.org/documentation/latest/authentication/ for details" % mesos_secrets)
         raise
-    return username, password
+    return Credentials(file=mesos_secrets, principal=username, secret=password)
+
+
+def get_principal(mesos_secrets='/nail/etc/mesos-slave-secret'):
+    """Helper function to get the principal from the mesos-slave credentials
+    :param mesos_secrets: optional argument specifying the path to the file containing the mesos-slave credentials
+    :returns: a string containing the principal/username
+    """
+    return load_credentials(mesos_secrets).principal
+
+
+def get_secret(mesos_secrets='/nail/etc/mesos-slave-secret'):
+    """Helper function to get the secret from the mesos-slave credentials
+    :param mesos_secrets: optional argument specifying the path to the file containing the mesos-slave credentials
+    :returns: a string containing the secret/password
+    """
+    return load_credentials(mesos_secrets).secret
+
+
+def reserve(slave_id, resources):
+    """Dynamically reserve resources in marathon to prevent tasks from using them.
+    :param slave_id: the id of the mesos slave
+    :param resources: list of Resource named tuples specifying the name and amount of the resource to (un)reserve
+    :returns: boolean where 0 represents success and 1 is a failure
+    """
+    log.info("Dynamically reserving resoures on %s: %s" % (slave_id, resources))
+    payload = {
+        'slaveId': slave_id,
+        'resources': str(build_reservation_payload(resources)).replace("'", '"').replace('+', '%20')
```

Comment:
Nope. The single quotes mess up the parsing and cause an `HTTPError: 400 Client Error: Bad Request`. In my opinion, this is a bug in Mesos itself.